packages/dd-trace/src/instrumenter.js

```diff
@@ -43,6 +43,22 @@ function getConfig (name, config = {}) {
   return config
 }
 
+function getPrepatchWrappedFunction (fn) {
+  const prepatchWrapped = function prepatchWrapped () {
+    const fnToCall = prepatchWrapped._datadog_wrapped || fn
+    if (new.target) {
+      // eslint-disable-next-line new-cap
+      return new fnToCall(...arguments)
+    } else {
+      return fnToCall.call(this, arguments)
+    }
+  }
+
+  // This will be an _actually_ wrapped function once this goes through Instrumenter#wrap
+  prepatchWrapped._datadog_wrapped = fn
```

Comment:
Where is the original stored when unpatching?

Reply:
It's kept in scope here as the argument to getPrepatchWrappedFunction.